module/x/gravity/keeper/grpc_query.go

```diff
@@ -307,6 +308,53 @@ func (k Keeper) ERC20ToDenom(c context.Context, req *types.ERC20ToDenomRequest)
 	return res, nil
 }
 
+func (k Keeper) DenomToERC20Params(c context.Context, req *types.DenomToERC20ParamsRequest) (*types.DenomToERC20ParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	metadata := k.bankKeeper.GetDenomMetaData(ctx, req.Denom) // lookup by base
+
+	if metadata.Base == "" { // not found, try to lookup by display
+		k.bankKeeper.IterateAllDenomMetaData(ctx, func(md banktypes.Metadata) bool {
+			if md.Display == req.Denom {
+				metadata = md
+				return true
+			}
+			return false
+		})
+	}
```

Comment:
What is this?

Reply:
It enables this UX improvement: 
https://github.com/PeggyJV/gravity-bridge/pull/106/files#r678716384

It is the main reason my ETH wallet shows 100k somm instead of 100k usomm.

Reply:
Correct me if I'm wrong @levicook, but I think this block of logic tries to find the metadata based on `base` OR `display`. i.e. if we can't find it by the `base`, we then try to find it by the `display`.

Reply:
That’s correct. And I would be okay with base only here (and everywhere else)

Reply:
Yea, I wouldn't put this in. I think it adds complexity to the code for very little benefit.

Reply:
> It is the main reason my ETH wallet shows 100k somm instead of 100k usomm.

I don't follow. Just to be clear, this adds a tiny bit of convenience for the contract deployer right?